src/ringbuffer.toit

```diff
@@ -15,24 +15,39 @@ class RingBuffer:
   append value/float:
     buffer[head] = value
     head = (head + 1) % size_
-    count = min (count + 1) size_
+    count = (min (count + 1) size_)
 
+  is_empty -> bool:
+    return count == 0
+
+  reduce [block]:
+    if is_empty: throw "Not enough elements"
+    result := null
+    is_first := true
+    for i := 0; i < count; i++:
+      if is_first: result = buffer[i]; is_first = false
```

Comment:
you could just set the result on instantiation to buffer[0] because it is not empty. that way you could remove the is_first boolean, right?

Reply:
Yes, but then you have to also adjust the indexing.  I derived this method from toit/lib/core/collections.toit line 2467 and so that Florian or others would recognize it instantly, did minimal changes.  Hope that is ok.